### PR TITLE
usb: Add cdc-demo app

### PR DESCRIPTION
**Pull request by MaciejPurski**
JIRA: RTOS-116

## Description

`cdc-demo` has been removed during phoenix-rtos-usb refactorization. Let's move it to phoenix-rtos-devices.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt106x).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: usb/libusbclient/imxrt-usbc/phy.c [hunk 1]
```diff
@@ -46,8 +46,8 @@ enum {
  * defined below, uncached map should be aligned to 0x800 both base and
  * end addresses
  */
-#define UNCACHED_BASE_ADDR     0x20020000
-#define UNCACHED_END_ADDR      0x20028000
+#define UNCACHED_BASE_ADDR 0x20050000
+#define UNCACHED_END_ADDR  0x20058000
```
Comment (agkaminski) [inline usb/libusbclient/imxrt-usbc/phy.c:50]: This change seems unrelated. Maybe this should be in separate PR/commit at least?

Comment (MaciejPurski) [reply usb/libusbclient/imxrt-usbc/phy.c:50]: I can put it in a different commit.
This change is needed in order to run USB client applications (e.g. cdc-demo) on imxrt without interfering with other processes, which use dtcm. The dtcm has been expanded recently in order to allow more processes to use it. Using hardcoded memory addresses is temporary, I hope.

**Review by agkaminski — APPROVED** [review]
